apis/v1beta1/shared_types.go

```diff
@@ -353,6 +355,46 @@ const (
 	RouteReasonBackendNotFound RouteConditionReason = "BackendNotFound"
 )
 
+const (
+	// This condition indicates that the Route contains a combination of both
+	// valid and invalid rules.
+	//
+	// When this happens, implementations MUST take one of the following
+	// approaches:
+	//
+	// 1) Drop Rule(s): With this approach, implementations will drop the
+	//    invalid Route Rule(s) until they are fully valid again. The message
+	//    for this condition MUST start with the prefix "Dropped Rule" and
+	//    include information about which Rules have been dropped. In this
+	//    state, the "Accepted" condition MUST be set to "True" with the latest
+	//    generation of the resource.
+	// 2) Fall Back: With this approach, implementations will fall back to the
+	//    last known good state of the entire Route. The message for this
+	//    condition MUST start with the prefix "Fall Back" and include
+	//    information about why the current Rule(s) are invalid. To represent
+	//    this, the "Accepted" condition MUST be set to "True" with the
+	//    generation of the last known good state of the resource.
+	//
+	// Reverting to the last known good state should only be done by
+	// implementations that have a means of restoring that state if/when they
+	// are restarted.
+	//
+	// This condition MUST NOT be set if a Route is fully valid, fully invalid,
+	// or not accepted.
+	//
+	// Possible reasons for this condition to be True are:
+	//
+	// * "UnsupportedValue"
+	//
+	// Possible reasons for this condition to be False are:
+	//
+	// * "PartiallyInvalid"
```

Comment:
```
type: PartiallyInvalid
status: False
reason: PartiallyInvalid
```

Can you explain this? "Its not PartiallyInvalid because of reason PartiallyInvalid"...?

Reply:
This is a good point - in this case, it probably makes sense to have the reason be "AllRulesValid" or something when the `status` is `False`, but we should also include guidance that the implementation MAY leave this Condition out if it would be set to `status: False`.

Reply:
Looks like above there is a statement:

```
	// This condition MUST NOT be set if a Route is fully valid, fully invalid,
	// or not accepted.
```

so maybe we can omit the guidance on `Possible reasons for this condition to be False are`?

Reply:
Thanks, good catch, removed entirely because it should never be set when it's false as @sunjayBhatia pointed out.

Reply:
(And also added a note that made it clear that the condition must not be set when it is false).